ouroboros-network/src/Ouroboros/Network/Protocol/LocalTxSubmission/Client.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RankNTypes          #-}
+
+
+-- | A view of the transaction submission protocol from the point of view of
+-- the client.
+--
+-- This provides a view that uses less complex types and should be easier to
+-- use than the underlying typed protocol itself.
+--
+-- For execution, 'localTxSubmissionClientPeer' is provided for conversion
+-- into the typed protocol.
+--
+module Ouroboros.Network.Protocol.LocalTxSubmission.Client (
+    -- * Protocol type for the client
+    -- | The protocol states from the point of view of the client.
+    LocalTxSubmissionClient(..)
+
+    -- * Execution as a typed protocol
+  , localTxSubmissionClientPeer
+  ) where
+
+import           Network.TypedProtocol.Core
+
+import           Ouroboros.Network.Protocol.LocalTxSubmission.Type
+
+
+-- | The client side of the local transaction submission protocol.
+--
+-- The peer in the client role submits transactions to the peer in the server
+-- role.
+--
+data LocalTxSubmissionClient tx m a where
+
+     SendMsgSubmitTx
+       :: tx
+       -> (Bool -> m (LocalTxSubmissionClient tx m a))
```

Comment:
^ maybe drop a comment line that this `Bool` parameter is the response from the server weather the previous transaction was valid or not.

Reply:
I've generalised it to include the rejection reason.